### ISSUE TITLE
train object detect model error

### ISSUE THREAD

**Issue by codeseeking** [tensorflow/models#7725]
I1028 14:33:08.725884 140177104160512 supervisor.py:1099] global_step/sec: 0
2019-10-28 14:33:17.640610: I tensorflow/core/kernels/data/shuffle_dataset_op.cc:111] Filling up shuffle buffer (this may take a while): 257 of 2048
2019-10-28 14:33:27.656747: I tensorflow/core/kernels/data/shuffle_dataset_op.cc:111] Filling up shuffle buffer (this may take a while): 416 of 2048
2019-10-28 14:33:37.657991: I tensorflow/core/kernels/data/shuffle_dataset_op.cc:111] Filling up shuffle buffer (this may take a while): 591 of 2048
2019-10-28 14:33:47.605458: I tensorflow/core/kernels/data/shuffle_dataset_op.cc:111] Filling up shuffle buffer (this may take a while): 1359 of 2048
2019-10-28 14:33:51.289321: I tensorflow/core/kernels/data/shuffle_dataset_op.cc:162] Shuffle buffer filled.
Killed
============================================================================
when I train ssd_mobile_v2, this problem is occuring,I dont know how to address it ,someone could tell me the resons how to result in ti?

**Comment by jneeven** [comment]
Looks like you might be running out of RAM. Could you try decreasing the shuffle buffer size from 2048 to some lower number?

**Comment by bozcani** [comment]
It seems that you select 2048 as buffer size which is too large for your memory. Can you try 1024? Simply, your code should look like this:
`train_dataset = train_dataset.shuffle(1024)`

**Comment by Wangweilai1** [comment]
I Have same problam, batch_size=1 and out of RAM

**Comment by bozcani** [comment]
@Wangweilai1 This error is not related to batch size. It is likely due to buffer size in tf.data.Dataset.shuffle() function:
https://github.com/tensorflow/tensorflow/blob/r2.0/tensorflow/python/data/ops/dataset_ops.py#L894-L921

Do you also use shuffle() function?

TF loads the number of _buffer_size_ items in the memory for shuffling. If your RAM capacity is not enough to hold the number of _buffer_size_ items, you can simply reduce it.

But you should be careful because lower values of _buffer_size_ may effect the randomness of your dataset. You can find here: https://stackoverflow.com/a/48096625

**Comment by ravikyram** [comment]
@codeseeking 

Is this still an issue?.Please, close this thread if your issue was resolved.Thanks!

**Comment by ravikyram** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
